numba_cuda/numba/cuda/cpython/cmathimpl.py

```diff
@@ -0,0 +1,566 @@
+"""
+Implement the cmath module functions.
+"""
+
+import cmath
+import math
+
+from numba.core.imputils import impl_ret_untracked
+from numba.core import types
+from numba.core.typing import signature
+from numba.cuda.cpython import mathimpl
+from numba.core.extending import overload
+from numba.cuda.cudaimpl import lower
+
+# ---------------------------------------------------------------------
+# XXX: In upstream Numba, this file would create a cmathimpl registry
+# if it was installed in the target (as it is for the CUDA target).
+# The cmathimpl registry has been removed from this file (it was
+# initialized as `registry = Registry('cmathimpl')`) as it would duplicate
+# the cmathimpl registry in upstream Numba, which would be likely to lead
+# to confusion / mixing things up between two cmathimpl registries. The
+# comment that accompanied this behaviour is left here, even though the
+# code that would pick the cmathimpl registry has been removed, for the
+# benefit of future understanding.
```

Comment:
I'm not sure if this comment is incorrect, or if I'm just not understanding it correctly. I don't see why creating a new registry here, and then installing it into the CUDA target would cause problems. Did you try this? If so, what issue do you run into?